### DISCUSSION TITLE
Issue with gas energy dashboard after upgrading from old version

### DISCUSSION THREAD

**Discussion by tom-pryor** [BottlecapDave/HomeAssistant-OctopusEnergy#1129] [Q&A]
Hello, I upgraded from what was probably a version at least over a year old and everything seems fine except the values for gas consumption in the energy dashboard. 

Previously I had the energy dashboard to use `sensor.octopus_energy_gas_{{METER_SERIAL_NUMBER}}_{{MPRN_NUMBER}}_current_consumption` which from the git history was the value in the docs until August 2023 (I would have set it up prior to this) and this worked fine. However, since updating this causes the energy dashboard to report massive (incorrect) kwh values for gas usage.

I can change the dashboard to use the current value in the docs of `sensor.octopus_energy_gas_{{METER_SERIAL_NUMBER}}_{{MPRN_NUMBER}}_current_accumulative_consumption_kwh` which then reports correct kwh values for gas but only from when I upgraded.

I've scanned the breaking changes in the releases and I can't see anything obvious. 

What is the proper migration for old setups using the previous `current_consumption` in the energy dashboard to preserve historical data?

Thanks!

**Comment by tom-pryor** [comment]
Hmm, maybe the `current_consumption` is a red herring? I see in this post about v10 https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/678 that the `sensor.octopus_energy_gas_{{METER_SERIAL_NUMBER}}_{{MPRN_NUMBER}}_current_accumulative_consumption` was split into m3 and kwh.

I tried the method in this discussion about renaming which sort of works and I now see my historical data and data from renaming but now I'm missing the data from when a upgraded to when I renamed the entity.

I deleted `sensor.octopus_energy_gas_{{METER_SERIAL_NUMBER}}_{{MPRN_NUMBER}}_current_accumulative_consumption` and renamed `sensor.octopus_energy_gas_{{METER_SERIAL_NUMBER}}_{{MPRN_NUMBER}}_current_accumulative_consumption_kwh` to `sensor.octopus_energy_gas_{{METER_SERIAL_NUMBER}}_{{MPRN_NUMBER}}_current_accumulative_consumption` but as mentioned I'm now missing the data in between the upgrade and rename. Is there some way to merge the data?

Would be good to know what the recommended migration procedure is, I've scanned the docs/FAQs and can't see anything. Clearly I must be doing something wrong as I'm still missing data.

**Reply by tom-pryor** [reply]
I'm guessing that thread about v10 is out of date as it doesn't work correctly as I noticed the following error when I renamed the entity which would explain the missing data:

```
Cannot rename statistic_id `sensor.octopus_energy_gas_xxx_xxx_current_accumulative_consumption_kwh` to `sensor.octopus_energy_gas_xxx_xxx_current_accumulative_consumption` because the new statistic_id is already in use
```

Anyway, in the end I used SQL to move the missing data in the database statistics tables from the `sensor.octopus_energy_gas_{{METER_SERIAL_NUMBER}}_{{MPRN_NUMBER}}_current_accumulative_consumption_kwh` entity to the `sensor.octopus_energy_gas_{{METER_SERIAL_NUMBER}}_{{MPRN_NUMBER}}_current_accumulative_consumption` and re-calculated the sums.

Seems to work but still be good to understand what I've done incorrectly here to end up in this state.

**Reply by BottlecapDave** [reply]
Hello. I'm afraid I don't really have the time to support people doing massive upgrade jumps. The recommended approach is to always upgrade one major version at a time, reading the changelogs. In this scenario, the suggestion is still applicable (to my knowledge), but I believe is time sensitive as it relies on the newly named entities not storing any data before it's renamed to avoid the clashes. I think SQL is unfortunately the only way in your scenario. 

Unfortunately, HA APIs don't seem to have a way to migrate entities using default names to new names, which caused this issue. I try to avoid entity name changes where possible.
